p2p/stream/common/streammanager/streamset.go

```diff
@@ -95,6 +96,20 @@ func (ss *streamSet) getStreams() []sttypes.Stream {
 	return res
 }
 
+func (ss *streamSet) popStream() (sttypes.Stream, error) {
+	ss.lock.RLock()
```

Comment:
i guess there should be `ss.lock.Lock()` because it modifies the struct.

Reply:
Nice catch, fixed, thanks